cmd/karmaExecuteTests.go

```diff
@@ -19,7 +19,7 @@ func karmaExecuteTests(config karmaExecuteTestsOptions, telemetryData *telemetry
 
 func runKarma(config karmaExecuteTestsOptions, command execRunner) {
 	installCommandTokens := tokenize(config.InstallCommand)
-	command.Dir(config.ModulePath)
+	command.SetDir(config.ModulePath)
```

Comment:
This change is not related to the scope of the pull request which is "Move mock capabilities ..." (applies also for other similar lines of code)

Reply:
But we assert the `dir` of a command. Therefore it has to be exported as `Dir` and the function needs to be renamed

Reply:
Also I needed to adapt all the files after moving the mock capabilities so that the tests will still run successfully

Reply:
I'm curious (and still in go ramp-up). Why do we have to replace `Dir` by `SetDir`? I would understand an update in case we have a e.g. `dir` or `setDir`(first char lower case), but why can't we use `Dir` in the same way like `SetDir`.

Reply:
for `karmeExecuteTests_test` we assert on the dir of the mockRunner.
To make that happen we have to export the dir from the mockRunner.

If we export Dir it has the same name as the function `Dir(d string)`. Therefore I have to rename the function. Since mockRunner and command have to implement the same interface I also renamed the function in `command`.

I would be happy to clear the confusion in a call tomorrow :)

Reply:
... understood. I reproduced it on my side by changing back to `Dir(./.)`.

With that we get:

> `pkg/mock/runner.go:36:6: type ExecMockRunner has both field and method named Dir`
>`pkg/mock/runner.go:62:6: type ShellMockRunner has both field and method named Dir`

Issue is resolved from my point of view.